### PR TITLE
Project Generator

### PR DESCRIPTION
**Pull request by pranavbaburaj**
- Added a project generator that generates hascal projects.
    
    `hascal init`
    This asks you a set of question, answer them all and you have a project
    created with a config file(`hascal.json`)

Another important thing to note:
I have used an external module call [`lol`](https://github.com/pranavbaburaj/lol). It is not really suggested to add it to the `core` folder(as it is over `20` files). 
So I have:
- Added the required packages inside of the `requirements.txt` file. 
- Added a `python_modules.bat` file that reads the `requirements.txt` and installs the packages. - Added this in the `build.bat` file(So the packages are automatically installed when building)

Please inform me if you find any problem with the code or please merge the commits

Hopefully Pranav

### CODE REVIEW

File: build.bat [hunk 1]
```diff
@@ -5,8 +5,8 @@ python --version
 echo =====================
 pip --version
 echo =====================
-echo Installing pyinstaller
-pip install pyinstaller
+echo Installing Packages
+pip install -r requirements.txt
```
Comment (pranavbaburaj) [inline build.bat:9]: This reads the requirements.txt file

File: docs/FirstApplication.md [hunk 2]
```diff
@@ -1 +1,20 @@
...
+```
+project
+│   hascal.json  
+│
+└───project
+│   │   main.has
+|  
+└───test
+    │   test.has
+```
```
Comment (pranavbaburaj) [inline docs/FirstApplication.md]: How the project generator works

**Comment by hasan-kashi** [comment]
Oh!it's very good.i will check your code and merge it into repo

**Comment by pranavbaburaj** [comment]
Thank you

**Comment by hasan-kashi** [comment]
I will merge your pull request ,but your code need to edit,we will edit it later

**Pull request merged**